pkg/kfconfig/types.go

```diff
@@ -516,7 +516,9 @@ func (c *KfConfig) SyncCache() error {
 				return errors.WithStack(err)
 			}
 		} else {
-			t := &http.Transport{}
+			t := &http.Transport{
+				Proxy: http.ProxyFromEnvironment,
```

Comment:
is there a way to provide a test for this?

Reply:
This can certainly be tested and the behavior is tested in detail here: https://github.com/golang/net/blob/master/http/httpproxy/proxy_test.go. However, I don't think it is valuable to test this in kfctl as it introduces undue maintenance burden on the kfctl developers. 

Using `http.ProxyFromEnvironment` is the default behavior for all HTTP request in Go([link](https://golang.org/src/net/http/transport.go#L42)) and this PR simply ensures the same behavior as the transport happens to be overriden.